sqlglot/expressions.py

```diff
@@ -4115,15 +4115,19 @@ def is_type(self, *dtypes: str | DataType | DataType.Type) -> bool:
         return self.to.is_type(*dtypes)
 
 
+class TryCast(Cast):
+    pass
+
+
 class CastToStrType(Func):
-    arg_types = {"this": True, "expression": True}
+    arg_types = {"this": True, "to": True}
 
 
-class Collate(Binary):
-    pass
+class CastToUserDefinedType(Func):
```

Comment:
why did you introduce a new cast? just use the regular one

Reply:
Because `Cast` also exposes `is_type`, which expects the `to` arg to be something that can be built into a `DataType`, so if a user called it with a UDT they would get unexpected `ParseError`s.

One workaround would be to enhance `Cast.is_type` so that it wraps the `to.is_type(..)` call in a try-except block and falls back to identifier comparison.

Reply:
make a new data type that is a custom one

Reply:
so make it the custom data types first class